modules/social_features/social_event/src/EdaHandler.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\social_event;
+
+use CloudEvents\V1\CloudEvent;
+use Drupal\Component\Uuid\UuidInterface;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\social_eda\DataFormatter as SocialEdaDataFormatter;
+use Drupal\social_eda_dispatcher\Dispatcher as SocialEdaDispatcher;
+use Drupal\social_event\Entity\Node\Event;
+use Drupal\social_event\Event\EventCreateEventData;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Eda Handler class.
+ */
+final class EdaHandler {
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(
+    private ?SocialEdaDispatcher $dispatcher,
+    private ?SocialEdaDataFormatter $formatter,
+    private UuidInterface $uuid,
+    private RequestStack $requestStack,
+  ) {}
+
+  /**
+   * Create event handler.
+   */
+  public function eventCreate(Event $event): void {
+    if ($this->dispatcher && $this->formatter) {
```

Comment:
If a condition spans the entire method, prefer using a [Guard Clause](https://en.wikipedia.org/wiki/Guard_(computer_science)) to improve readability.

Also prefer explicit NULL checks for clarity.

```suggestion
    if ($this->dispatcher === NULL || $this->formatter === NULL) {
      return;
    }
```